src/rust/engine/src/nodes.rs

```diff
@@ -212,17 +220,17 @@ impl Select {
     variant_value: Option<&str>
   ) -> Option<Value> {
     // Check whether the subject is-a instance of the product.
-    if let Some(&candidate) = self.select_literal_single(context, candidate, variant_value) {
-      return Some(candidate)
+    if let Some(&ref candidate) = self.select_literal_single(context, candidate, variant_value) {
```

Comment:
Is it just same as "...Some(candidate)..."?
My understanding is select_literal_single returns Option<&Value>, thus Some(X), X is already &Value type.
Some(&X) makes X be Value type, Some(&ref X) makes X &Value type again.